server/redeemer.go

```diff
@@ -0,0 +1,383 @@
+			if err != nil {
+				glog.Error(err)
+			}
+			if err := stream.Send(&net.MaxFloatUpdate{MaxFloat: maxFloat.Bytes()}); err != nil {
+				glog.Error(err)
+				if err == io.EOF {
+					return err
+				}
+			}
+		}
+	}
+}
+
+type RedeemerClient struct {
+	conn *grpc.ClientConn
+	rpc  net.TicketRedeemerClient
+
+	senders map[ethcommon.Address]*remoteSender
+	mu      sync.RWMutex
+	quit    chan struct{}
+	sm      pm.SenderManager
+	tm      pm.TimeManager
+}
+
+type remoteSender struct {
+	maxFloat   *big.Int
+	stream     net.TicketRedeemer_MonitorMaxFloatClient
+	done       chan struct{}
+	lastAccess time.Time
+}
+
+// NewRedeemerClient instantiates a new client for the ticket redemption service
+// The client implements the pm.SenderMonitor interface
+func NewRedeemerClient(uri *url.URL, sm pm.SenderManager, tm pm.TimeManager) (*RedeemerClient, error) {
+	conn, err := grpc.Dial(
+		uri.String(),
+		grpc.WithBlock(),
+		grpc.WithTimeout(GRPCConnectTimeout),
+		grpc.WithInsecure(),
+	)
+
+	//TODO: PROVIDE KEEPALIVE SETTINGS
+	if err != nil {
+		glog.Errorf("Did not connect to orch=%v err=%v", uri, err)
+		return nil, fmt.Errorf("Did not connect to orch=%v err=%v", uri, err)
+	}
+	return &RedeemerClient{
+		conn:    conn,
+		rpc:     net.NewTicketRedeemerClient(conn),
+		sm:      sm,
+		tm:      tm,
+		senders: make(map[ethcommon.Address]*remoteSender),
+		quit:    make(chan struct{}),
+	}, nil
+}
+
+func (r *RedeemerClient) Start() {
+	r.startCleanupLoop()
+}
+
+// Stop stops the Redeemer client
+func (r *RedeemerClient) Stop() {
+	close(r.quit)
+	r.conn.Close()
+}
+
+// QueueTicket sends a winning ticket to the Redeemer
+func (r *RedeemerClient) QueueTicket(ticket *pm.SignedTicket) error {
+	ctx, cancel := context.WithTimeout(context.Background(), GRPCTimeout)
+	defer cancel()
+	_, err := r.rpc.QueueTicket(ctx, protoTicket(ticket))
+	return err
+}
+
+// MaxFloat returns the max float for 'sender'
+// If no local cache is available this method will remotely request
+// max float from the Redeemer server and start watching for subsequent updates from the Redeemer server
+func (r *RedeemerClient) MaxFloat(sender ethcommon.Address) (*big.Int, error) {
+	// Ensure thread-safe access to 'RedeemerClient.senders'
+	r.mu.Lock()
+
+	// Check whether local cache exists for 'sender'
+	if mf, ok := r.senders[sender]; ok && mf.maxFloat != nil {
+		r.senders[sender].lastAccess = time.Now()
+		return mf.maxFloat, nil
+	}
+
+	// Retrieve max float from Redeemer and cache it if no local cache exists
+	ctx, cancel := context.WithTimeout(context.Background(), GRPCTimeout)
+	defer cancel()
+	mfu, err := r.rpc.MaxFloat(ctx, &net.MaxFloatReq{Sender: sender.Bytes()})
+	if err != nil {
+		return nil, err
+	}
+	mf := new(big.Int).SetBytes(mfu.MaxFloat)
+
+	// Request updates for sender from Redeemer
+	stream, err := r.rpc.MonitorMaxFloat(ctx, &net.MaxFloatReq{Sender: sender.Bytes()})
+	// By sending a message over the existing stream to the Redeemer
```

Comment:
Not sure that this comment is accurate - the client isn't sending a message over the stream to the Redeemer here?

Reply:
fixed in https://github.com/livepeer/go-livepeer/pull/1513/commits/6b7ac892cd933ab613b66a1cd5ae30cd990c128e